accounts-db/src/accounts_db.rs

```diff
@@ -2862,6 +2862,19 @@ impl AccountsDb {
                                         } else {
                                             found_not_zero += 1;
                                         }
+
+                                        // If this candidate has multiple rooted slot list entries,
+                                        // we should reclaim the older ones.
+                                        if slot_list.len() > 1
+                                            && *slot
+                                                <= max_clean_root_inclusive.unwrap_or(Slot::MAX)
```

Comment:
If max_clean_root_inclusive is none, we need to follow the old rule, i.e. slots must be in self.uncleaned_roots...

Reply:
but following the old rule may lead to less effective clean (which was achieved by this PR) ... 

In other words, you can't get the benefit of this clean if you have `snapshot disabled` on your validator?

if we follow the old rule, snapshot disabled + in memory index will have the problem of account's map growth?
if we follow the new rule, snapshot disabled nodes will clean incorrectly?

Reply:
@HaoranYi @brooksprumo did we decide this is not a concern because of the discussion here:
https://github.com/anza-xyz/agave/pull/4044/files#r1881080106